### DISCUSSION TITLE
How would I specify a higher version devDependency than a coinciding peerDepedency?

### DISCUSSION THREAD

**Discussion by dkershner6** [projen/projen#3219] [Q&A] [answered]
Let's say I would like to install `constructs` as `10.3.0`, but place a peerDependency for my package as `^10`, how would I do so?

I have tried some obvious choices like `@^10` and `@>=10` for peerDeps with `@10.3.0` or `@^10.3.0` for devDeps, but it always installs `10.0.0`.

Ironically, I just got a warning from JSII that I shouldn't be doing this, so I am assuming this was put in place as a JSII limitation.

**Comment by mrgrain** [comment, marked as answer]
There's a setting to disable to pinned dev dependency.

The default behavior happens for a reason. If you specify a peer dependency of `^10`, a user of your package can install `10.0.0` and that would be correct. If you however locally use `10.3.0`, you might accidentally depend on a feature that is not available in `10.0.0` yet and thus unintentionally break the `^10` contract.

**Reply by dkershner6** [reply]
Perfect, thank you.  I see it.

For others:
```typescript

export interface PeerDependencyOptions {
  /**
   * Automatically add a pinned dev dependency.
   * @default true
   */
  readonly pinnedDevDependency?: boolean;
}
```

I understand your concern, and it is more valid than mine: That a package will break backwards compatibility, but this does happen fairly frequently to us, usually through external dependency interactions, like the AWS SDK.  I just needed a way to override and test, and this provides that.  Thank you.
